docs/consensus/consensus_chain.md

```diff
@@ -106,6 +106,16 @@ Justifications contain a set of all PoT checkpoints since the parent block up to
 4. DSN sync must be able to terminate early if local chain already contains imported blocks that DSN sync was about to download (doesn’t happen often, but possible)
 5. Node blocks that are finalized and pruned must be much higher than archiving point such that block available through DSN sync and regular Substrate sync have significant overall (5 archived segments worth of blocks right now)
 
+## Fast sync
+
+1. Obtain segment headers from DSN: 1-4 from `sync from DSN implementation`
+2. If the last segment index is greater than 2 - download the last two segments from DSN otherwise exit `Fast sync` and continue with either `sync from DSN implementation` or `Substrate sync`.
+3. Reconstruct blocks from the two downloaded segments.
```

Comment:
Can you remind me why do we need two segments instead of just latest one? I thought padding block with zeroes removed the need to download an extra segment.

Reply:
We need to "force import" one block and the whole segment after that to keep the archiver valid. Potentially, we can get a single block by a special block request (not publicly exposed in Substrate yet) but I'm not sure the "padding optimization" alone would be enough.

Reply:
Why wouldn't it be? We literally throw away extra bytes at the beginning anyway: https://github.com/subspace/subspace/blob/2d1e4dc6a44d088c774aee8735c5e035948c8023/crates/subspace-archiving/src/archiver.rs#L329

Reply:
We likely could use the padding mechanism you described however we need the block from the previous segment to "force import".  And since we have the block already it makes sense to use it as it is.

Reply:
I thought about it some more and even following that logic why would we download two segments if we can download just one and then import just last full block of the last segment and switch to regular sync from there? Switching to sync from DSN also doesn't make a lot of sense since we already use latest segment, we can switch to Substrate sync immediately.

Reply:
In theory, we need three "special" blocks - one for "forced import", a block to instantiate an archiver from the correct point (first of the segment), and a block for "NetworkBroadcast". Maybe we can reuse either the first two or the last two but we need an additional block.

> Switching to sync from DSN also doesn't make a lot of sense since we already use the latest segment, we can switch to Substrate sync immediately.

Also, after we downloaded a segment there could be a new segment or (segments) already.

Reply:
> In theory, we need three "special" blocks - one for "forced import", a block to instantiate an archiver from the correct point (first of the segment), and a block for "NetworkBroadcast". Maybe we can reuse either the first two or the last two but we need an additional block.

Theoretically we just need exacty one: to force import the first block of the beginning of the new segment. Everything else after that will "just work". ver already knows how to initialize from such state, we just need to teach it to "re-initialize" itself if it had block N and then suddenly got block N+100500 and some blocks in between are missing, while corresponding segment headers are already stored.

> Also, after we downloaded a segment there could be a new segment or (segments) already.

Full nodes intentionally store a few segments worth of blocks locally, even in somewhat rare case when this happens (due to how little we need to download) this will not be an issue in practice.